pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/mercury/MercuryOrderDao.java

```diff
@@ -171,10 +173,19 @@ public HashMap<String, MercuryOrderDto> getPossibleBarcodesForParticipant(String
 
             return dbVals;
         });
+
+        if (results.resultException != null) {
+            throw new RuntimeException("Error getting the mercury orders", results.resultException);
```

Comment:
Thanks Elliot. But where exactly will this exception be caught and reported?

Reply:
Thx. I read the call stack incorrectly earlier. I see the route now.

Is there translation of RuntimeException to a specific HTTP status code in Spark? 
In this case, is this a request error or an internal error?

Reply:
Any that aren't handled directly are turned into a 500.  In this case, it will present as a 500 to the user.

Reply:
Is this an internal error or a problem with the request?

Reply:
The best answer for this case is: yes - it's either a request error or an internal one.  It's likely a case of a bad request, but there's at least one edge case I can see that bubbles up where it wouldn't be: the database goes down during this part of the request/response lifecycle.
The only thing this addition is doing is surfacing the error instead of returning a possibly incomplete map.
What would you like to see happen here?  We could continue with the pattern as-is, change it for this case, or change it for all of the cases.

Reply:
Thx for the explanation Elliot.
I'd like us to return an appropriate response code from all our routes, but I'm starting to realize that's a big job. So for now if we know the code we want to return (for code under change or development) we should try to align exceptions with response codes. (I see this, but I am not sure if there is a better approach: http://sparkjava.com/documentation.html#error-handling. I have not used Spark before -- at least not this Spark.)
If there is not a clean opportunity to quickly fix those issues here, of course go ahead with the merge. I appreciate what Pegah is trying to accomplish with the PR.